service/src/prover_node.rs

```diff
@@ -1,17 +1,27 @@
 use once_cell::sync::OnceCell;
-use std::sync::Mutex;
+use std::sync::{Arc, Mutex};
 
 #[derive(Debug, Clone)]
 pub struct ProverNode {
     pub addr: String,
+    pub client: Arc<Mutex<Option<tonic::transport::channel::Channel>>>,
 }
 
 impl ProverNode {
     pub fn new(addr: &String) -> Self {
         ProverNode {
             addr: addr.to_string(),
+            client: Arc::new(Mutex::new(None)),
         }
     }
+
+    pub fn get_client(&self) -> Option<tonic::transport::channel::Channel> {
+        self.client.lock().unwrap().clone()
+    }
+
+    pub fn set_client(&mut self, client: Option<tonic::transport::channel::Channel>) {
```

Comment:
It's better to init the client in the function `new`.

Reply:
Will set it to none when some error happened,  so need set_client

Reply:
Okay, you are designing a stateful ProverNode, and the client can be switched/attached to a different client/connection.  So you can change the `set_client` to switch_to or attach. Meanwhile.